CMSgov/mint-app
# [EASI-3320] -  Move Possible Solution Points of Contact to the backend

# EASI-3320

## Changes and Description

This feature extracted hard-coded solution contacts from the FE and now store them in the BE.  This allows us to leverage that information when sending emails to recipients.  The BE exposes this data now through a GQL query to map into existing FE solution data

### FE Changes
- Hook to fetch possible solution contacts and map them to existing FE solution map
- Removed unneeded hard-coded solution contacts from existing map
- Updated unit tests to implement this async hook integration

### BE Changes
- Backend changes were reviewed and tested in a [separate PR](https://github.com/CMSgov/mint-app/pull/730)
   - New Contact Type
       - Data loader
       - Resolvers
   - Additional Solutions added to the possible operational solutions table 



## How to test this change

- Log into MINT and navigate to the help center for solutions
- Verify contacts are all fetched and rendered, both on the cards and the modal
- Navigate to the IT tracker and select an operational need (will need to be triggered by a question)
- Verify the contact is listed on the checkbox card
- Continue the flow for the IT tracker
- Verify the contact is rendered on all variations of the solution cards and modal

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
